### DISCUSSION TITLE
Trying to boot iPXE from another server

### DISCUSSION THREAD

**Discussion by derFuchsi** [ipxe/ipxe#223] [Q&A]
Hi there! I hope you can help me with my special problem. I searched the web but can't find a working solution. 
(Pls forgive my possibly poor english)

I have 2 seperate preinstalled servers with different software starting over PXE. These servers worked fine as single bootservers for a while, but the idea now is (not my idea, i just have the problem) to get them working at the same network now (without changing them much) so the clients can choose where to boot from. So i built a 3rd server with **Ubuntu** and made a pxe boot menu with **dnsmasq**. I had a little trouble to find out how to get that working (i'm neither a Linux or an network crack) but now it runs fine with the one server booting **pxelinux**.  The other server boots **iPXE** and when i boot from the server directly it works fine.  But when i boot from the menu iPXe starts but then searches for `http://10.77.80.2/iPXE.menu` instead of `http://10.77.80.1/iPXE.menu` running into an error "could not chain iPXE menu"

dnsmasq.conf:
```
Interface=enp11s0f1
bind-interfaces
domain=linuxhint.local

dhcp-range=enp11s0f1,10.77.80.4,10.77.80.255,255.0.0.0,8h
dhcp-option=option:router,10.77.80.2
dhcp-option=option:dns-server,10.77.80.2
dhcp-boot=pxelinux.0,linuxhint-s20,10.77.80.2
dhcp-no-override

enable-tftp
tftp-no-fail
tftp-root=/netboot/tftp

pxe-prompt="Choose bootserver"
pxe-service=x86PC,"local boot",pxelinux
pxe-service=x86PC, "iPXE Win Server", Boot\iPXE\undionly.kpxe, 10.77.80.1
pxe-service=x86PC, "Pxelinux Suse Server", Boot/pxelinux, 10.77.80.3
```

10.77.80.2 is the Ubuntu boot sever
10.77.80.3 client boots fine
10.77.80.1 client boots but the software runs into the error described above

Can i avoid the error by modifying anything at my Ubuntu Server?

**Comment by NiKiZe** [comment]
Dnsmasq, and at least one other machine serves as a DHCP server, or DHCP proxy server.
In general having more than one DHCP server on the same network is to ask for trouble, the same goes for having multiple DHCP proxy servers.

I would recommend using tcpdump or wireshark to inspect the traffic and understanding the DHCP packets. But essentially what you see here is that your new Ubuntu proxy DHCP responds several times, the first when PXEClient runs and that gives the first menu, then again when iPXE runs, and at this time you will want the server to respond with the settings used for "the old iPXE" server.

**Reply by derFuchsi** [reply]
Yes i now have 3 DHCP servers but that's not my fault ^^. The task is i just have to get it working without changing too much on the original servers if possible. I know it's dirty but as long as the ubuntu server responds first (and it does) all is fine. The problem is i don't know how to tell iPXE the right IP. Maybe it's just an option in dnsmasq?  I read something about  **next-server**...?
Is there an iPXE config file? I can't find one on the iPXE server.

**Reply by NiKiZe** [reply]
You did setup the 3rd one didn't you?
You have a setup that works more by chance than anything else.

iPXE does a DHCP discover that has all the settings, the DHCP server could/should be configured to handle those and respond differently, or even better, not at all.

You also have the option to embed a script into the iPXE binary.

Again you are doing something really bad here!

**Reply by derFuchsi** [reply]
Yes the 3rd one is from me. And it works well with the other server running pxelinux.

I tried in my test setup here with deactivated DHCP on the iPXE server - doesn't make any difference. The DHCP on the iPXE server has nothing to do with this error.

**Reply by derFuchsi** [reply]
I figured it out now. No wizardry necessary here, i just had to add the proxy option `dhcp-range=10.77.80.2,proxy` to dnsmasq.
Sorry for wasting your time, i'm just bumbling around here ^^. Maybe this helps someone some day.
